scripts/generateCliCode.ts

```diff
@@ -0,0 +1,348 @@
+    for (const propertyName of Object.keys(preferences.properties)) {
+      this.walkProperty(propertyName, preferences.properties[propertyName], this.settingsTree);
+    }
+  }
+
+  protected async emitOutput(outputFile: string) {
+    const options = { rmWhitespace: false };
+    const templateFile = 'scripts/assets/options.go.templ';
+
+    const linesForJSON = this.collectServerSettingsForJSON(this.settingsTree, true, '');
+    const linesWithoutJSON = this.collectServerSettingsForJSON(this.settingsTree, false, '');
+    const data = {
+      fieldsToUpdate:   this.fieldsToUpdate,
+      linesForJSON:     linesForJSON.join('\n'),
+      linesWithoutJSON: linesWithoutJSON.join('\n'),
+    };
+    const renderedContent = await ejs.renderFile(templateFile, data, options).then((rc: string) => rc);
+
+    if (!renderedContent) {
+      throw new Error('ejs.renderFile returned nothing');
+    }
+    if (outputFile === '-') {
+      console.log(renderedContent);
+
+      return;
+    }
+
+    const parent = path.dirname(outputFile);
+
+    try {
+      fs.accessSync(parent, fs.constants.W_OK | fs.constants.X_OK);
+    } catch (ex: any) {
+      if (ex.code === 'ENOENT') {
+        console.log(`Creating directory ${ parent }...`);
+        fs.mkdirSync(parent, { recursive: true });
+      } else {
+        throw ex;
+      }
+    }
+    fs.writeFileSync(outputFile, renderedContent);
+  }
+
+  protected collectServerSettingsForJSON(settingsTree: Record<string, any>, includeJSONTag: boolean, indent: string): string[] {
+    return Object.keys(settingsTree).map((propertyName) => {
+      return this.collectServerSettingsForJSONProperty(propertyName, settingsTree[propertyName], includeJSONTag, indent);
+    }).flat();
+  }
+
+  protected collectServerSettingsForJSONProperty(propertyName: string, settingsTree: Record<string, any>, includeJSONTag: boolean, indent: string): string[] {
+    const decln = `${ indent }${ propertyName } `;
+
+    if (typeof (settingsTree.type) === 'object') {
+      const lines: string[] = [];
+
+      lines.push(`${ decln } struct {`);
+      lines.push(...this.collectServerSettingsForJSON(settingsTree.type, includeJSONTag, `${ indent }  `));
+      const parts = [indent, '}'];
+
+      if (includeJSONTag) {
+        parts.push(` \`json:"${ uncapitalize(propertyName) }"\``);
+      }
+      lines.push(parts.join(''));
+
+      return lines;
+    }
+    const parts = [decln, ' '];
+
+    if (includeJSONTag) {
+      parts.push('*');
+    }
+    parts.push(settingsTree.type);
+    if (includeJSONTag) {
+      parts.push(`\`json:"${ uncapitalize(propertyName) },omitempty"\``);
+    }
+
+    return [parts.join('')];
+  }
+
+  protected convertStringsToGolang(enums: string[] | undefined): string {
+    return !enums ? '' : `[]string{${ enums.map(s => JSON.stringify(s) ).join(', ') }}`;
+  }
+
+  protected getCommandLineArgValue(lcTypeName: string, capitalizedName: string) {
```

Comment:
```suggestion
  protected getCommandLineArgValue(lcTypeName: leafTypeValue, capitalizedName: string) {
```
(Assuming the proposed type definition above)

This would also let us drop the `default` case, since the compiler will ensure that will never be reached.

Reply:
This has been done.